### PR TITLE
[MSFT] Finally move away from GlobalRefOp

### PR DESCRIPTION
**Pull request by teqdruid**
Switch to HierPathOp. Closes #5183.

### CODE REVIEW

**Comment by teqdruid** [comment]
@seldridge I see there's a verbatim substitution for HierPathOp, but it outputs `.` for the separator. I need `|`. The obvious solution to this is to add format options to the substitution string (e.g. `{{1:|}}`. Do you think that's appropriate? Could be useful for other substitutions.

**Comment by seldridge** [comment]
Yeah, the format options makes sense to me.

File: lib/Dialect/MSFT/Transforms/MSFTLowerInstances.cpp [hunk 1]
```diff
@@ -166,21 +114,6 @@ void LowerInstancesPass::runOnOperation() {
...
-    SmallVector<Attribute> newGlobalRefs(
-        llvm::map_range(refArr, [](hw::GlobalRefAttr ref) { return ref; }));
-    Operation *op = opRefPair.getFirst();
-    if (auto refArr =
-            op->getAttrOfType<ArrayAttr>(hw::GlobalRefAttr::DialectAttrName))
-      newGlobalRefs.append(refArr.getValue().begin(), refArr.getValue().end());
-    op->setAttr(hw::GlobalRefAttr::DialectAttrName,
-                ArrayAttr::get(ctxt, newGlobalRefs));
-  }
 }
```
Comment (seldridge) [inline lib/Dialect/MSFT/Transforms/MSFTLowerInstances.cpp:117]: I'm glad to see that this is removing a bunch of code. 😉

**Review by seldridge — APPROVED** [review]
LGTM

File: lib/Dialect/MSFT/Transforms/MSFTLowerInstances.cpp (hunk 1 above)
Comment (teqdruid) [reply lib/Dialect/MSFT/Transforms/MSFTLowerInstances.cpp:117]: Same.

**Pull request merged**